### ISSUE TITLE
Only compile_train threads batch-norm write-backs, so predict returns garbage

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor-ml#104]
Batch norm's running statistics are written by the op and collected only by `compile_train`, so anyone who assembles their own updates and compiles with `function` trains to convergence and then predicts against `mean=0, var=1`.

```python
import numpy as np
import pytensor.tensor as pt

from pytensor_ml.layers import BatchNorm, Linear, Sequential
from pytensor_ml.model import Model
from pytensor_ml.optim import adam
from pytensor_ml.pytensorf import function

x, y = pt.matrix("x"), pt.vector("y")
net = Sequential(Linear("fc1", 3, 4), BatchNorm("bn", 4), Linear("fc2", 4, 1))
prediction = net(x)[:, 0]
model = Model(x, prediction).initialize(seed=0)
loss = pt.mean((prediction - y) ** 2)

step = function([x, y], loss, updates=adam(1e-1)(loss, model.weights))

rng = np.random.default_rng(0)
features = rng.normal(size=(64, 3)) * 10.0 + 5.0  # off unit scale, so frozen stats bite
targets = features @ np.arange(3.0)
for _ in range(200):
    step(features, targets)

print(float(step(features, targets)))                                  # 7.9e-07
print(float(np.mean((model.predict(features) - targets) ** 2)))        # 494104.8
# fix: step = compile_train(loss, adam(1e-1), inputs=[x, y])  -> predict 5.2e-07
```
